sqlglot/parser.py

```diff
@@ -5517,16 +5518,18 @@ def _parse_id_var(
         self,
         any_token: bool = True,
         tokens: t.Optional[t.Collection[TokenType]] = None,
+        comment: bool = False,
     ) -> t.Optional[exp.Expression]:
-        identifier = self._parse_identifier()
-        if identifier:
-            return identifier
-
-        if (any_token and self._advance_any()) or self._match_set(tokens or self.ID_VAR_TOKENS):
+        expression = self._parse_identifier()
+        if not expression and (
+            (any_token and self._advance_any()) or self._match_set(tokens or self.ID_VAR_TOKENS)
+        ):
             quoted = self._prev.token_type == TokenType.STRING
-            return exp.Identifier(this=self._prev.text, quoted=quoted)
+            expression = exp.Identifier(this=self._prev.text, quoted=quoted)
+            if comment:
+                self._add_comments(expression)
```

Comment:
It's handled automatically because `_parse_identifier` constructs an expression using `self.expression`, which in turn takes care of attaching comments:

```python
return self.expression(exp.Identifier, this=self._prev.text, quoted=True)
```